engine/src/main/java/org/terasology/rendering/opengl/ScreenGrabber.java

```diff
@@ -112,36 +117,70 @@ public void saveScreenshot() {
         int width = sceneFinalFbo.width();
         int height = sceneFinalFbo.height();
 
-        Runnable task = () -> {
-            SimpleDateFormat sdf = new SimpleDateFormat("yyMMddHHmmss");
-
-            final String format = renderingConfig.getScreenshotFormat();
-            final String fileName = "Terasology-" + sdf.format(new Date()) + "-" + width + "x" + height + "." + format;
-            Path path = PathManager.getInstance().getScreenshotPath().resolve(fileName);
-            BufferedImage image = new BufferedImage(width, height, BufferedImage.TYPE_INT_RGB);
-
-            for (int x = 0; x < width; x++) {
-                for (int y = 0; y < height; y++) {
-                    int i = (x + width * y) * 4;
-                    int r = buffer.get(i) & 0xFF;
-                    int g = buffer.get(i + 1) & 0xFF;
-                    int b = buffer.get(i + 2) & 0xFF;
-                    image.setRGB(x, height - (y + 1), (0xFF << 24) | (r << 16) | (g << 8) | b);
-                }
-            }
-
-            try (OutputStream out = new BufferedOutputStream(Files.newOutputStream(path))) {
-                ImageIO.write(image, format, out);
-                logger.info("Screenshot '" + fileName + "' saved! ");
-            } catch (IOException e) {
-                logger.warn("Failed to save screenshot!", e);
-            }
-        };
+        Runnable task;
+        if (savingGamePreview) {
+            task = () -> saveGamePreviewTask(buffer, width, height);
```

Comment:
... what dark magic is this? Probably this is totally legit and fine I've just never seen it before :D What does it even do to have a `()` in place like that?

Reply:
I think it's one of these: https://hashnode.com/post/anonymous-functions-in-java-explained-cj1opkbj8000sml53bsq6r6cj